gunicorn/workers/gtornado.py

```
@@ -48,12 +46,29 @@ def watchdog(self):
 
         if self.ppid != os.getppid():
             self.log.info("Parent changed, shutting down: %s", self)
-            self.stop()
+            self.alive = False
+
+    def heartbeat(self):
+        if not self.alive:
+            if self.server_alive:
+                if hasattr(self, 'server'):
```

The exception is ignored,but in fact we don't know how to deal with it.LGTM.